lambda/src/newtab-app-lambda-edge/__tests__/newtab-lambda-edge.test.js

```diff
@@ -81,4 +81,82 @@ describe('newtab app Lambda@Edge function on origin-request', () => {
     const request = callback.mock.calls[0][1]
     expect(request.uri).toEqual('/newtab/profile/stats/something.txt')
   })
+
+  it('does not change the origin when there are no cookies passed in request.headers.cookie', () => {
+    const { handler } = require('../newtab-app-lambda-edge')
+    const event = getMockCloudFrontEventObject()
+    event.Records[0].cf.request.headers.cookie = []
+    const originalOrigin = event.Records[0].cf.request.origin
+    const context = getMockLambdaContext()
+    handler(event, context, callback)
+    const request = callback.mock.calls[0][1]
+    expect(request.origin).toMatchObject(originalOrigin)
+  })
+
+  it('does not change the origin when there are unrelated cookies passed in request.headers.cookie', () => {
+    const { handler } = require('../newtab-app-lambda-edge')
+    const event = getMockCloudFrontEventObject()
+    event.Records[0].cf.request.headers.cookie = []
+    event.Records[0].cf.request.headers.cookie = [
+      { key: 'Cookie', value: 'something=here' },
+      { key: 'Cookie', value: 'foo=bar' },
+    ]
+    const originalOrigin = event.Records[0].cf.request.origin
+    const context = getMockLambdaContext()
+    handler(event, context, callback)
+    const request = callback.mock.calls[0][1]
+    expect(request.origin).toMatchObject(originalOrigin)
+  })
+
+  it('changes the origin to ZEIT Now when the v4 beta opt-in cookie is passed in request.headers.cookie', () => {
+    const { handler } = require('../newtab-app-lambda-edge')
+    const event = getMockCloudFrontEventObject()
+    event.Records[0].cf.request.headers.cookie = [
+      { key: 'Cookie', value: 'something=here' },
+      { key: 'Cookie', value: 'tabV4OptIn=enabled' },
+    ]
+    const context = getMockLambdaContext()
+    handler(event, context, callback)
+    const request = callback.mock.calls[0][1]
+    expect(request.origin).toMatchObject({
+      custom: {
+        domainName: 'tab-web.now.sh',
```

Comment:
TODO: this should be an env var, because we want different endpoints for development.